src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -232,21 +232,47 @@ export class TerminalPanel extends Panel {
 					return;
 				}
 
-				// Check if the file was dragged from the tree explorer
-				let uri = e.dataTransfer.getData('URL');
+				const winFormatters: [RegExp, (uri: URI) => string][] = [
```

Comment:
Let's pull this and the application of the formatter into the `_preparePathForTerminal` function.